plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/SpringReload.java

```diff
@@ -0,0 +1,572 @@
+package org.hotswap.agent.plugin.spring;
+
+import org.hotswap.agent.logging.AgentLogger;
+import org.hotswap.agent.plugin.spring.core.*;
+import org.hotswap.agent.plugin.spring.files.PropertyReload;
+import org.hotswap.agent.plugin.spring.getbean.ProxyReplacer;
+import org.hotswap.agent.plugin.spring.utils.AnnotatedBeanDefinitionUtils;
+import org.hotswap.agent.plugin.spring.files.XmlReload;
+import org.hotswap.agent.util.AnnotationHelper;
+import org.hotswap.agent.util.spring.util.ClassUtils;
+import org.hotswap.agent.util.spring.util.ObjectUtils;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.annotation.AnnotatedGenericBeanDefinition;
+import org.springframework.beans.factory.config.*;
+import org.springframework.beans.factory.support.*;
+import org.springframework.core.type.MethodMetadata;
+import org.springframework.core.type.StandardMethodMetadata;
+
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.hotswap.agent.plugin.spring.utils.RegistryUtils.maybeRegistryToBeanFactory;
+import static org.hotswap.agent.util.ReflectionHelper.get;
+
+/**
+ * Reload spring beans.
+ */
+public class SpringReload {
```

Comment:
Maybe we should rename this class as 'BeanfactoryReloader'.

Reply:
Rename as SpringBeanReload